src/platforms/javascript/common/session-replay/configuration.mdx

```diff
@@ -87,21 +88,57 @@ new Replay({
+You can use the Sentry SDK to link a user to a session. Read more about it in our <PlatformLink to="/enriching-events/identify-user">docs</PlatformLink>.
+
+```javascript
+Sentry.setUser({ email: "jane.doe@example.com" });
+```
+
+## Using a Custom Compression Worker
+
+By default, Session Replay will use an inlined web worker script to compress Replay data before we send it over the network. This helps to drastically reduce the amount of data transferred, improving performance and reducing network overhead. Because compressing data is CPU intensive, we use a web worker to offload this work to a separate thread.
+
+While an inlined worker will work well for most applications, there are two main problems with this:
+
+1. Since the worker code is inlined, it increases the main bundle size of your application.
+2. Using inline workers can cause CSP violations if your application has a very strict CSP policy. [Learn more about configuring your CSP policy.](../#content-security-policy-csp)
+
+To solve these problems, we provide the ability to use a custom worker script. This allows you to host the worker script on your own domain, and avoid the CSP issues that come with inline workers. Additionally, the worker script can then be removed from the main application bundle, and can be served and cached separately.
+
+Follow these steps in order to use a custom compression worker:
+
+1. Download the [example worker script](https://github.com/getsentry/sentry-javascript/blob/develop/packages/replay-worker/examples/worker.min.js) from our Github repository.
+2. Host this worker script on the same domain as your application. For example, if your application is hosted at `https://example.com`, you could host the worker script at `https://example.com/assets/worker.min.js`.
+3. Configure your custom worker script in replay:
 
 ```javascript
 new Replay({
-  mutationBreadcrumbLimit: 1000,
-  mutationLimit: 1500,
+  workerUrl: "/assets/worker.min.js",
 });
 ```
 
-## Identifying Users
+4. You can now [tree shake](../../configuration/tree-shaking/) the default included worker script. If you are using our bundler plugins (version 2.10.0 or later), you can do:
 
-You can use the Sentry SDK to link a user to a session. Read more about it in our <PlatformLink to="/enriching-events/identify-user">docs</PlatformLink>.
+```javascript
+sentryPlugin({
+  // other config
+  bundleSizeOptimizations: {
+    excludeReplayWorker: true,
+  },
+});
+```
+
+Please note that since you are hosting this script yourself, you are responsible for keeping it up to date. We recommend that you check for updates to the worker script periodically, and update it as needed. The worker script API is guaranteed to remain stable inside of a major version, so any v7 worker script should be backwards and forwards compatible with any v7 SDK version. However, we may make improvements or bug fixes to the worker script, which you may miss out if you don't update it.
```

Comment:
```suggestion
Since you are hosting this script yourself, you are responsible for keeping it up to date. We recommend that you check for updates to the worker script periodically, and update it as needed. The worker script API is guaranteed to remain stable inside of a major version, so any v7 worker script should be backward and forward compatible with any v7 SDK version. However, we may make improvements or bug fixes to the worker script, which you will miss out on if you don't update it.
```